requirements.txt

```diff
@@ -1,11 +1,13 @@
 numpy==1.26.0; platform_machine != 'armv7l'
 numpy==1.25.2; platform_machine == 'armv7l'
-pandas==2.0.3
+pandas==2.1.1
```

Comment:
Pandas was reverted intentionally, and the version 2.1.1 is not one we will be able to use (it contains a bug that causes tons of (false) deprecation warnings.

We'll update pandas when it's ready - but surely not as part of this PR.

Reply:
yes, of course, this PR won't work as intended before that.

Reply:
I'm not entirely sure what would've changed in a 2.1.1 that influences this PR? 

Aside from that - Then we'll have to postpone looking at this further, at least until 2.1.2 is released.

The problematic bug:
https://github.com/pandas-dev/pandas/issues/55025
a comment near the end promises this to be included in 2.1.2.

The problem is visible if you run pytest (or a strategy) - causing tons of deprecation warnings due to string assignments - which are however creating the column - causing a wrong warning.

Reply:
before `2.1.1`, using `pd.to_numeric` in some scenarios has a chance of losing meaningful decimals. This has been addressed in `2.1.1`.

Reply:
unfortunate :laughing: 

all 2.1.x have that string column problem - and each "deprecation" warning they issue causes a slight delay.

Reply:
well, `as_type()` could be used to do the same thing but downcasting needs a bunch of validation to prevent precision loss. I think relying on `pd.to_numeric()` would be better in this case because it takes some considerations into account.

Reply:
hi @AliSayyah 
pandas finally released the 2.1.2 version, which is what we now depend on.

with that, i think we can start looking at this actively again.

Reply:
Hi @xmatthias 
Thank you for reminding me of the release. yes, I agree. How should we proceed?

Reply:
probably
fix conflict
fix ci

then i'll have a spin to see if this works well for all intended usecases, or needs some change / adjustment?

Reply:
Hey @xmatthias, got caught up with mid-terms but I've sorted out those conflicts. Can you please give the CI workflow a thumbs up?